pytest_base_url/plugin.py

```diff
@@ -21,13 +24,21 @@ def base_url(request):
 def _verify_url(request, base_url):
     """Verifies the base URL"""
     verify = request.config.option.verify_base_url
+    verify_timeout = request.config.getoption('verify_base_url_attempts')
     if base_url and verify:
-        response = requests.get(base_url, timeout=10)
-        if not response.status_code == requests.codes.ok:
-            raise pytest.UsageError(
-                'Base URL failed verification!'
-                '\nURL: {0}, Response status code: {1.status_code}'
-                '\nResponse headers: {1.headers}'.format(base_url, response))
+        session = requests.Session()
+        retries = Retry(total=verify_timeout,
+                        backoff_factor=0.1,
+                        status_forcelist=[500, 502, 503, 504])
+        session.mount(base_url, HTTPAdapter(max_retries=retries))
+        try:
+            session.get(base_url)
+        except Exception:
```

Comment:
Are you saying to try and get which exact error code it failed on? I don't think there is a way to get it since once it errors it throws that ```ConnectionError``` I posted above.